test/extended/util/configv1shim_test.go

```diff
@@ -0,0 +1,276 @@
+package util
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	configv1 "github.com/openshift/api/config/v1"
+	applyconfigv1 "github.com/openshift/client-go/config/applyconfigurations/config/v1"
+	fakeconfigv1client "github.com/openshift/client-go/config/clientset/versioned/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func createInfrastructureObject(name string) *configv1.Infrastructure {
+	return &configv1.Infrastructure{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "config.openshift.io/v1",
+			Kind:       "Infrastructure",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{},
+		},
+		Spec: configv1.InfrastructureSpec{
+			PlatformSpec: configv1.PlatformSpec{
+				Type: configv1.AWSPlatformType,
+			},
+		},
+		Status: configv1.InfrastructureStatus{
+			APIServerInternalURL:   "https://api-int.jchaloup-20230222.group-b.devcluster.openshift.com:6443",
+			APIServerURL:           "https://api.jchaloup-20230222.group-b.devcluster.openshift.com:6443",
+			ControlPlaneTopology:   configv1.HighlyAvailableTopologyMode,
+			EtcdDiscoveryDomain:    "",
+			InfrastructureName:     "jchaloup-20230222-cvx5s",
+			InfrastructureTopology: configv1.HighlyAvailableTopologyMode,
+			Platform:               configv1.AWSPlatformType,
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+				AWS: &configv1.AWSPlatformStatus{
+					Region: "us-east-1",
+				},
+			},
+		},
+	}
+}
+
+func TestConfigClientShimErrorOnMutation(t *testing.T) {
+	updateNotPermitted := OperationNotPermitted{Action: "update"}
+	updatestatusNotPermitted := OperationNotPermitted{Action: "updatestatus"}
+	patchNotPermitted := OperationNotPermitted{Action: "patch"}
+	applyNotPermitted := OperationNotPermitted{Action: "apply"}
+	applyStatusNotPermitted := OperationNotPermitted{Action: "applystatus"}
+	deleteNotPermitted := OperationNotPermitted{Action: "delete"}
+	deleteCollectionNotPermitted := OperationNotPermitted{Action: "deletecollection"}
+
+	object := createInfrastructureObject("cluster")
```

Comment:
maybe name them staticObject and realObject/dynamicObject? or something else for better recognition?

Reply:
Done